econml/ortho_forest.py

```diff
@@ -55,11 +55,29 @@ def _build_tree_in_parallel(Y, T, X, W,
 
 
 def _fit_weighted_pipeline(model_instance, X, y, sample_weight):
+    weights_error_msg = (
+        "Estimators of type {} do not accept weights. "
+        "Consider using the class WeightedModelWrapper from econml.utilities to build a weighted model."
+    )
+    expected_error_msg = "fit() got an unexpected keyword argument 'sample_weight'"
     if not isinstance(model_instance, Pipeline):
-        model_instance.fit(X, y, sample_weight)
+        try:
+            model_instance.fit(X, y, sample_weight=sample_weight)
+        except TypeError as e:
+            if expected_error_msg in str(e):
```

Comment:
Would that not be very sensitive to the exact error msg that sklearn raises here? Is there a more robust way to catch this?